### PR TITLE
Embed youtube video demo-ing CI/CD pipeline

### PR DESCRIPTION
**Pull request by yorinasub17**
Since the video is actually 35 seconds, and I didn't know a good tool for converting mp4 to gif, I decided to upload the demo into a youtube video and embed it in asciidoc.

The link to the video is here: https://youtu.be/iYXghJK7YdU

### CODE REVIEW

**Comment by josh-padnick** [comment]
Hey Yori, I just viewed the video and I must admit this was pretty fast and pretty hard to follow. We've put a ton of work into thinking through the framework, implementing it, and now writing this deployment guide, so I wonder if it's worth you doing a more paced demo with audio narration to give the user context on what they're looking at, let the screen linger on the same text for a few seconds while you chat, and walk the user through our overall thought process?

**Comment by yorinasub17** [comment]
Well, I was trying to get it under 30 seconds to turn into a gif for easier embedding and rendering for the announcement blog post, but with the number of steps in the pipeline, it was difficult 😅

I think this video works for a teaser in an announcement blog post, but you are right that it makes the solution here cheap if embedded in the production deployment guide.

I'll see what I can do here.

**Comment by brikis98** [comment]
Thanks for putting this together Yori!

I agree with Josh the video is a bit hard to follow. I think one improvement that should be straightforward to do in, for example, iMovie, is to add text overlays explaining what's happening in each step.

Also, in terms of the script, here's how I think about it:

| What's happening in the video                                        | Text overlay                                                      |
|----------------------------------------------------------------------|-------------------------------------------------------------------|
| Making changes to `terragrunt.hcl` in a text editor                  | Make a change in your Terraform code.                             |
| Typing `git commit -m "Deploy version..."` in a terminal             | Commit your changes and open a pull request.                      |
| Pull request in GitHub, click on CircleCi build to see `plan` output | Automated tests and `terraform plan` run automatically.           |
| Click "Merge" on pull request in GitHub.                             | Merge the pull request to master.                                 |
| Notification pops up in Slack.                                       | CI system automatically runs `terraform plan` and notifies admin. |
| Scrolling through `plan` output in CircleCi.                         | Admin reviews `terraform plan` output in CI server.               |
| Admin allows build to continue in CircleCi.                          | Admin approves the `plan`.                                        |
| `terraform apply` running in CircleCi.                               | CI system automatically runs `terraform apply`.                   |
| Show "apply complete" output at end of Terraform run.                | Your new Terraform code is now deployed!                          |

**Comment by yorinasub17** [comment]
@josh-padnick @brikis98 Ok I recorded a new video. This one is ~5 mins long and has audio commentary from me. It's not the best production as I fumbled on the words a few times, but I think it gets the job done.

Let me know if anything should be polished!

**Comment by brikis98** [comment]
@yorinasub17 The video looks great! I took it and also created a short, 1-minute gif version with subtitles:


Gruntwork Terraform Pipeline

We could include the gif as the "TLDR" at the top and your full video a bit later in the post (and same thing with the deployment guide) for the full details?

**Comment by yorinasub17** [comment]
Gif looks awesome! Thanks for doing that - I could have done it myself!

Just included it at the top, and then moved my video to the deployment walkthrough section. Do you think that placement is correct?

**Comment by yorinasub17** [comment]
Also updated the blog post: https://blog.gruntwork.io/how-to-configure-a-production-grade-ci-cd-workflow-for-infrastructure-code-15c50bea5ae7

Note that I only added the gif in the post, and then linked to the Youtube video in the captions.

File: _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc [hunk 1]
```diff
@@ -29,6 +29,11 @@ Delivery pipeline for your infrastructure code. This guide will walk you through
 pipeline for your favorite infrastructure as code tools (e.g., Terraform) using your favorite CI/CD platform (e.g.,
 Jenkins, Circle, GitLab, etc).
 
+Here is what the pipeline looks like in action:
+
+image::/assets/img/guides/infrastructure-cicd-pipeline/walkthrough.gif[]
```
Comment (brikis98) [inline _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc]: NIT: consider adding a sentence to tell the user there's a longer, more detailed video walkthrough below, plus the guide will expose all the details

File: _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc [hunk 2]
```diff
@@ -29,6 +29,11 @@ Delivery pipeline for your infrastructure code. This guide will walk you through
 pipeline for your favorite infrastructure as code tools (e.g., Terraform) using your favorite CI/CD platform (e.g.,
 Jenkins, Circle, GitLab, etc).
 
+Here is what the pipeline looks like in action:
```
Comment (brikis98) [inline _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc]:
```suggestion
TLDR: If you follow this guide, you'll be able to set up a pipeline that works like this:
```

File: _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc (hunk 1 above)
Comment (yorinasub17) [reply _posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc]: I decided to link to the youtube video in the caption just like with the blog post and remove it from the bottom.

**Comment by yorinasub17** [comment]
Thanks! Merging in now.
